### PR TITLE
Issue #2409059: Porting Send Mail Functionality (continues PR #390)

### PR DESCRIPTION
**Pull request by emclaughlin1215**
Issue #2409059: Fixing up the tests so that they pass and fixing up the action so it works. Other actions with logging do not have translatable log messages, so I opted to remove that, which made testing easier.

Following up on work done on #390

### CODE REVIEW

**Comment by emclaughlin1215** [comment]
Failure was Drupal 8.1.x on PHP 7 in code I didn't touch? Relevant debug info:

> There was 1 error:
> 1) Drupal\Tests\rules\Functional\UiPageTest::testCreateReactionRule
> Behat\Mink\Exception\DriverException: There is no element matching XPath "//html"
> /home/travis/build/fago/drupal/vendor/behat/mink-browserkit-driver/src/BrowserKitDriver.php:832
> /home/travis/build/fago/drupal/vendor/behat/mink-browserkit-driver/src/BrowserKitDriver.php:347
> /home/travis/build/fago/drupal/vendor/behat/mink/src/Element/Element.php:176
> /home/travis/build/fago/drupal/vendor/behat/mink/src/WebAssert.php:257
> /home/travis/build/fago/rules/tests/src/Functional/UiPageTest.php:76
> FAILURES!
> Tests: 289, Assertions: 957, Errors: 1.

Not sure what to do about this. Suggestions?

File: tests/src/Integration/Action/SystemMailToUsersOfRoleTest.php [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    // @todo this is wrong, the logger is no factory.
+    $this->container->set('logger.factory', $logger_factory->reveal());
+    $this->container->set('plugin.manager.mail', $this->mailManager->reveal());
+    $config = [
+      'system.site' => [
+        'mail' => $test_email,
+      ],
+    ];
+    $this->container->set('config.factory', $this->getConfigFactoryStub($config));
+    $this->container->set('entity_type.manager', $this->entityTypeManager->reveal());
```
Comment (klausi) [inline tests/src/Integration/Action/SystemMailToUsersOfRoleTest.php]: this is already done in one of the parent classes, so the line can be removed.

File: tests/src/Integration/Action/SystemMailToUsersOfRoleTest.php [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+   */
+  public function providerSendMailToRole() {
+    // Create mock roles here instead of in setUp because this runs before
+    // setUp.
+    foreach (['administrator', 'editor'] as $role_name) {
+      // Mock the role entity.
+      $role_mock = $this->prophesize(RoleInterface::class);
+      $role_mock->id()
+        ->willReturn($role_name);
+      $roles[] = $role_mock->reveal();
```
Comment (klausi) [inline tests/src/Integration/Action/SystemMailToUsersOfRoleTest.php]: $roles is never used? Why do you set it up here?

**Comment by klausi** [comment]
continued in #408

**Pull request closed without merging**